### PR TITLE
MAINT: resolve pyflake F403	'from module import *' used

### PR DESCRIPTION
**Pull request by mwtoews**
* For external modules, resolve imported members
* Most internal relative modules were ignored or marked `noqa: F403`
* Convert a few internal absolute imports to relative imports

Found using:
```
flake8 --select=F403 | grep -v "'from \." | sort
```
the regular expression ignores internal relative imports

### CODE REVIEW

**Comment by mattip** [comment]
not sure we need the extra comment `# noqa: F403`, but I guess it doesn't hurt and will make additional cleanup easier to find

File: numpy/matlib.py [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import numpy as np
 from numpy.matrixlib.defmatrix import matrix, asmatrix
-# need * as we're copying the numpy namespace (FIXME: this makes little sense)
-from numpy import *
+# need * as we're copying __all__ objects from the numpy namespace
+from numpy import *  # noqa: F403
```
Comment (eric-wieser) [inline numpy/matlib.py]: Why did we we remove the `FIXME` here?

File: numpy/distutils/ccompiler.py [hunk 2]
```diff
@@ -7,9 +7,10 @@
 import subprocess
 from copy import copy
 from distutils import ccompiler
-from distutils.ccompiler import *
+from distutils.ccompiler import compiler_class, gen_lib_options, \
+    get_default_compiler, new_compiler, CCompiler
```
Comment (eric-wieser) [inline numpy/distutils/ccompiler.py]: Can you use parens instead of backslashes here?

Comment (mwtoews) [reply numpy/distutils/ccompiler.py]: done

File: numpy/matlib.py (hunk 1 above)
Comment (mwtoews) [reply numpy/matlib.py]: I was hoping to improve the comment to say why `from numpy import *` is needed in this context. Does this comment help, or can it be further improved?

**Review by mwtoews — COMMENTED** [review]
> not sure we need the extra comment `# noqa: F403`, but I guess it doesn't hurt and will make additional cleanup easier to find

For reference, see [flake8's dcoumentation on "In-line Ignoring Errors"](http://flake8.pycqa.org/en/3.1.1/user/ignoring-errors.html#in-line-ignoring-errors). For any users of flake8, this message will normally be hidden for this line of code. It is also my mark to acknowledge and accept the use of `from module import *`, which is generally discouraged.

File: numpy/core/setup.py [hunk 3]
```diff
@@ -14,7 +14,7 @@
     uses_accelerate_framework, get_sgemv_fix
     )
 from numpy.compat import npy_load_module
-from setup_common import *
+from setup_common import *  # noqa: F403
```
Comment (mwtoews) [inline numpy/core/setup.py]: For the record, I had difficulties making this into either a relative or absolute import. I can't recall exactly, but I think I tried `from .setup_common import *` and `from numpy.core.setup_common import *`, but both bombed Travis CI tests. If anyone knows why, I'd be interested.

File: numpy/matlib.py (hunk 1 above)
Comment (mwtoews) [reply numpy/matlib.py]: It seems this comment is further addressed in #15381, so I'm pulling this edit out of this PR.

**Comment by mattip** [comment]
Putting this in. I think we should limit the code churn simply to change relative/absolute imports. Both styles are commonly used, and it is not worth the effort to unify the code base.

**Pull request merged**